### PR TITLE
add helper for closing instructions overlay

### PR DESCRIPTION
**Pull request by Hamms**
Also a general helper for clicking on an element if it exists.

### CODE REVIEW

File: dashboard/test/ui/step_definitions/steps.rb [hunk 1]
```diff
@@ -73,6 +73,12 @@ def install_js_error_recorder
   @browser.switch_to.window(@browser.window_handles.last)
 end
 
+When /^I close the instructions overlay if it exists$/ do
+  steps <<-STEPS
+    When I click selector ".csf-top-instructions button:contains(OK)" if it exists
+  STEPS
```
Comment (islemaster) [inline dashboard/test/ui/step_definitions/steps.rb]: As a one-liner you could skip the HEREDOC.

File: dashboard/test/ui/step_definitions/steps.rb [hunk 2]
```diff
@@ -295,6 +301,12 @@ def install_js_error_recorder
   @browser.execute_script("$(\"#{jquery_selector}\")[0].click();")
 end
 
+When /^I click selector "([^"]*)" if it exists$/ do |jquery_selector|
+  if @browser.execute_script("return (\"#{jquery_selector}\").length > 0")
```
Comment (islemaster) [inline dashboard/test/ui/step_definitions/steps.rb]: Should there be a `$` in there somewhere?

Comment (Hamms) [reply dashboard/test/ui/step_definitions/steps.rb]: .... huh. How was this passing?

Comment (islemaster) [reply dashboard/test/ui/step_definitions/steps.rb]: Well, the string length is always `> 0`.  Are we ever hitting the negative case where the thing you might click is not there?

Comment (Hamms) [reply dashboard/test/ui/step_definitions/steps.rb]: Ah! We are, but only in eyes tests, which don't run on circle.

**Comment by davidsbailey** [comment]
👍

**Pull request merged**